### PR TITLE
Fix return value of undo() and redo() methods.

### LINKED ISSUE #460: Wrong return value of undo() method
**Describe the bug**
In some cases `undo()` method returns wrong result.

**To Reproduce**
1.  Add a text.
2. Add a brush drawing
3. Call `undo()`
4. `undo()` returns a false although there is a text drawing that can be undone.

**Expected behavior**
`undo()` call returns true so text also can be removed.

**Additional context**
While looking into code I found this:

```
    fun undoView(): Boolean {
        if (mViewState.addedViewsCount > 0) {
            val removeView = mViewState.getAddedView(
                mViewState.addedViewsCount - 1
            )
            if (removeView is DrawingView) {
                return removeView.undo()
            } else {
                mViewState.removeAddedView(mViewState.addedViewsCount - 1)
                mViewGroup.removeView(removeView)
                mViewState.pushRedoView(removeView)
            }
            when (val viewTag = removeView.tag) {
                is ViewType -> onPhotoEditorListener?.onRemoveViewListener(
                    viewTag,
                    mViewState.addedViewsCount
                )
            }
        }
        return mViewState.addedViewsCount != 0
    }
  ```
  I think ` return removeView.undo()` causes the issue as it doesn't count added views before `DrawingView`.

### PR DESCRIPTION
**Pull request by VardanTitan**
Fixes #460

### CODE REVIEW

**Comment by VardanTitan** [comment]
@burhanrashid52 Feel free to modify changes. Also, wasn't sure about versioning.

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/GraphicManager.kt [hunk 1]
```diff
@@ -75,7 +75,7 @@ internal class GraphicManager(
                 mViewState.redoViewsCount - 1
             )
             if (redoView is DrawingView) {
-                return redoView.redo()
+                return redoView.redo() && (mViewState.redoViewsCount != 0)
```
Comment (wasky) [inline photoeditor/src/main/java/ja/burhanrashid52/photoeditor/GraphicManager.kt:78]: Redo still doesn't work as expected. Steps to reproduce:

* Add a brush drawing
* Add a text
* Call `undo()` twice
* Calling `redo()` returns `false`. Should return `true`.

Comment (VardanTitan) [reply photoeditor/src/main/java/ja/burhanrashid52/photoeditor/GraphicManager.kt]: @wasky I changed the condition and also changed the redo logic a bit. Consider a scenario when the user does a redo action and draws another thing on top of it. I think we should clear redo history in this case.

**Review by wasky — APPROVED** [review]
Looks good to me, but I'm not very familiar with this part of the code in the lib.

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/GraphicManager.kt (hunk 1 above)
Comment (burhanrashid52) [reply photoeditor/src/main/java/ja/burhanrashid52/photoeditor/GraphicManager.kt]: Yeah, I think it's LGTM as well, also the CI UI test is passing so I am assuming everything is working fine. Thanks for review and testing it out @wasky

**Comment by burhanrashid52** [comment]
Thanks, @VardanTitan for the PR now it is live at [3.0.2](https://central.sonatype.com/artifact/com.burhanrashid52/photoeditor/3.0.2?smo=true) and also @wasky for the review. Let me know what you think! Thanks.

**Pull request merged**